lib/tools/node-modules/node-modules-dest-copy.ts

```diff
@@ -51,6 +51,30 @@ export class TnsModulesCopy {
 
 			// remove platform-specific files (processed separately by plugin services)
 			shelljs.rm("-rf", path.join(targetPackageDir, "platforms"));
+
+			this.removeNotProductionDependencies(dependency, targetPackageDir);
+		}
+	}
+
+	private removeNotProductionDependencies(dependency: IDependencyData, targetPackageDir: string): void {
+		let packageJsonFilePath = path.join(dependency.directory, "package.json");
+		if (!this.$fs.exists(packageJsonFilePath)) {
+			return;
+		}
+
+		let packageJsonContent = this.$fs.readJson(packageJsonFilePath);
+		let productionDependencies = packageJsonContent.dependencies;
+
+		let dependenciesFolder = path.join(targetPackageDir, "node_modules");
+		if (this.$fs.exists(dependenciesFolder)) {
+			let dependencies = this.$fs.readDirectory(dependenciesFolder);
+
+			for (let i = 0; i < dependencies.length; i++) {
```

Comment:
The `for` loop can be replaced with

```typescript
let dependencies = this.$fs.readDirectory(dependenciesFolder);
dependencies.filter(dir => !!productionDependencies || !productionDependencies.hasOwnProperty(dir))
             .map(dir => shelljs.rm("-rf", path.join(dependenciesFolder, dir));
```

for consistency with the rest of the code base.